lib/multi_channel.ml

```diff
@@ -32,8 +32,11 @@ type dls_state = {
   mc: mutex_condvar;
 }
 
+module Foreign_queue = Lockfree.Michael_scott_queue
+
 type 'a t = {
   channels: 'a Ws_deque.t array;
+  foreign_queue: 'a Foreign_queue.t;
```

Comment:
Is `foreign_queue` the queue where domainslib tasks unblocked by an external library end up? Some code comments would improve readability.

Reply:
Yes, that is the idea.